### PR TITLE
[REF] wowl: make localization a service

### PR DESCRIPTION
**Pull request by brboi**

### CODE REVIEW

File: addons/wowl/static/src/services/user.ts [hunk 1]
```diff
@@ -73,13 +72,7 @@ export const userService: Service<UserService> = {
...
 
     return {
-      dateFormat: localization.dateFormat,
-      decimalPoint: localization.decimalPoint,
-      direction: localization.direction,
-      grouping: localization.grouping,
-      multiLang: localization.multiLang,
-      thousandsSep: localization.thousandsSep,
-      timeFormat: localization.timeFormat,
+      ...env.services.localization,
```
Comment (ged-odoo) [inline addons/wowl/static/src/services/user.ts]: remove this

File: addons/wowl/static/src/env.ts [hunk 2]
```diff
@@ -4,21 +4,26 @@ import { OdooConfig, OdooEnv } from "./types";
...
-    _t,
     debug,
+  } as OdooEnv;
+
+  const translateFn: PropertyDescriptor = {
+    get() {
+      return env.services.localization._t;
+    },
   };
+  Object.defineProperty(env, "_t", translateFn);
```
Comment (ged-odoo) [inline addons/wowl/static/src/env.ts]: move this in localization service

Comment (brboi) [inline addons/wowl/static/src/env.ts]: move to localization service

File: addons/wowl/static/src/env.ts [hunk 3]
```diff
@@ -4,21 +4,26 @@ import { OdooConfig, OdooEnv } from "./types";
...
+  Object.defineProperty(qweb, "translateFn", translateFn);
```
Comment (ged-odoo) [inline addons/wowl/static/src/env.ts]: to do in localization service
```js
env.qweb.translateFn = ...
```

File: addons/wowl/static/src/env.ts [hunk 4]
```diff
@@ -4,21 +4,26 @@ import { OdooConfig, OdooEnv } from "./types";
 import { OdooBrowser } from "./types";
 
 export async function makeEnv(config: OdooConfig): Promise<OdooEnv> {
-  const { templates, _t, debug } = config;
-  const qweb = new owl.QWeb({ translateFn: _t });
-  qweb.addTemplates(templates);
+  const { templates, debug } = config;
+  const qweb = new owl.QWeb({ templates });
```
Comment (ged-odoo) [inline addons/wowl/static/src/env.ts]:
```js
  const qweb = new owl.QWeb({ templates, translateFn: () => throw new Error("BOOM" });

```

File: addons/wowl/static/src/env.ts [hunk 5]
```diff
@@ -1,24 +1,21 @@
...
-  const qweb = new owl.QWeb({ translateFn: _t });
-  qweb.addTemplates(templates);
-
+export async function makeEnv(debug: string): Promise<OdooEnv> {
   const env: OdooEnv = {
     browser: owl.browser,
-    qweb,
+    qweb: new owl.QWeb({
+      translateFn: () => {
+        throw new Error("BOOM");
```
Comment (ged-odoo) [inline addons/wowl/static/src/env.ts]: maybe need a better error message :)

File: addons/wowl/static/src/services/localization.ts [hunk 6]
```diff
@@ -109,3 +109,12 @@ _t("about a month ago");
 _t("%d months ago");
 _t("about a year ago");
 _t("%d years ago");
+
+export const localizationService: Service<Localization> = {
+  name: "localization",
+  async deploy(env: OdooEnv): Promise<Localization> {
+    const localization = await fetchLocalization();
+    env._t = env.qweb.translateFn = localization._t;
```
Comment (ged-odoo) [inline addons/wowl/static/src/services/localization.ts]: i would prefer that in 2 different lines

File: addons/wowl/static/src/utils/fields_utils.ts [hunk 7]
```diff
@@ -153,21 +153,21 @@ function strftime_to_moment_format(value: any) {
  * Get date format of the user's language
  */
 export function getLangDateFormat(env: OdooEnv) {
```
Comment (ged-odoo) [inline addons/wowl/static/src/utils/fields_utils.ts]: move this and the following into localization service

Comment (ged-odoo) [reply addons/wowl/static/src/utils/fields_utils.ts]: (maybe)

File: addons/wowl/static/src/services/localization.ts [hunk 8]
```diff
@@ -109,3 +109,12 @@ _t("about a month ago");
 _t("%d months ago");
 _t("about a year ago");
 _t("%d years ago");
+
+export const localizationService: Service<Localization> = {
+  name: "localization",
+  async deploy(env: OdooEnv): Promise<Localization> {
+    const localization = await fetchLocalization();
```
Comment (ged-odoo) [inline addons/wowl/static/src/services/localization.ts]: make boom function in env._t

File: addons/wowl/static/src/main.ts [hunk 9]
```diff
@@ -51,16 +50,10 @@ const { whenReady, loadFile } = utils;
...
   // setup environment
-  const env = await makeEnv({
-    localization,
-    debug: odoo.debug!,
-    templates,
-    _t,
-  });
+  const [env, templates] = await Promise.all([makeEnv(odoo.debug!), loadTemplates()]);
+  env.qweb.addTemplates(templates);
+  WebClient.env = env;
```
Comment (ged-odoo) [inline addons/wowl/static/src/main.ts]: remove this

**Pull request merged**